edit/linter-meta.js

```diff
@@ -0,0 +1,48 @@
+/* global linter */
+'use strict';
+
+function createMetaCompiler(cm) {
+  const successCallbacks = [];
+  let meta = null;
+  let cache = [];
+
+  linter.register((text, options, _cm) => {
+    if (_cm !== cm) {
+      return;
+    }
+    const match = text.match(/\/\*\s*==userstyle==[\s\S]*?==\/userstyle==\s*\*\//i);
+    if (!match) {
+      return [];
+    }
+    if (match[0] === meta) {
+      return cache;
+    }
+    return parseMeta(match[0])
+      .then(result => {
+        for (const cb of successCallbacks) {
+          cb(result);
+        }
+        meta = match[0];
+        cache = [];
+        return cache;
+      }, err => {
+        meta = match[0];
+        cache = [{
+          from: cm.posFromIndex((err.index || 0) + match.index),
```

Comment:
I think @tophf had made an optimized version of `posFromIndex`. Is it possible to use it here? e.g.  `cm.betterPosFromIndex(...)`.